qiskit_experiments/curve_analysis/composite_curve_analysis.py

```diff
@@ -280,54 +332,31 @@ def _run_analysis(
     ) -> Tuple[List[AnalysisResultData], List["matplotlib.figure.Figure"]]:
 
         analysis_results = []
+        figures = []
```

Comment:
This is almost copy of `CurveAnalysis` class, but will be completely dropped once after we move `ScatterTable` to artifact. Eventually I want to write

```python
component_analysis_results = []
for analysis in self._analyses:
    analysis.set_options(plot=False)
    sub_results, _ = analysis._run_analysis(experiment_data)
    component_analysis_results.extend(sub_results)

# --- I need this logic ---
# 1. get ScatterData of the sub analysis from the artifact
# 2. combine sub scatter data
figures = self._create_figures(combined_scatter_data)

return component_analysis_results + composite_results
```

but currently scatter data is discarded. Note that composite curve analysis usually creates a single plot including curves from sub analysis, c.f. HamiltonianTomo. Current implementation is just to keep scatter data of the sub analyses, so do not need to review carefully.